src/types/payload.ts

```diff
@@ -1,236 +1,12 @@
-  releases_url: Type.String(),
-  size: Type.Number(),
-  ssh_url: Type.String(),
-  stargazers_count: Type.Number(),
-  stargazers_url: Type.String(),
-  statuses_url: Type.String(),
-  subscribers_url: Type.String(),
-  subscription_url: Type.String(),
-  svn_url: Type.String(),
-  tags_url: Type.String(),
-  teams_url: Type.String(),
-  topics: Type.Array(Type.Any()),
-  trees_url: Type.String(),
-  updated_at: Type.String({ format: "date-time" }),
-  url: Type.String(),
-  visibility: Type.String(),
-  watchers_count: Type.Number(),
-  watchers: Type.Number(),
-  web_commit_signoff_required: Type.Boolean(),
-});
-
-const organizationSchema = Type.Object({
-  login: Type.String(),
-  id: Type.Number(),
-  node_id: Type.String(),
-  url: Type.String(),
-  repos_url: Type.String(),
-  events_url: Type.String(),
-  hooks_url: Type.String(),
-  issues_url: Type.String(),
-  members_url: Type.String(),
-  public_members_url: Type.String(),
-  avatar_url: Type.String(),
-  description: Type.Union([Type.String(), Type.Null()]),
-});
-
-const installationSchema = Type.Object({
-  id: Type.Number(),
-  node_id: Type.String(),
-});
-
-const commentSchema = Type.Object({
-  author_association: Type.String(),
-  body_html: Type.Optional(Type.String()),
-  body_text: Type.Optional(Type.String()),
-  body: Type.String(),
-  created_at: Type.String({ format: "date-time" }),
-  html_url: Type.String(),
-  id: Type.Number(),
-  issue_url: Type.String(),
-  node_id: Type.String(),
-  updated_at: Type.String({ format: "date-time" }),
-  url: Type.String(),
-  user: userSchema,
-  reactions: Type.Object({
-    url: Type.String(),
-    total_count: Type.Number(),
-    "+1": Type.Number(),
-    "-1": Type.Number(),
-    laugh: Type.Number(),
-    hooray: Type.Number(),
-    confused: Type.Number(),
-    heart: Type.Number(),
-    rocket: Type.Number(),
-    eyes: Type.Number(),
-  }),
-});
-
-const changesSchema = Type.Object({
-  body: Type.Optional(
-    Type.Object({
-      from: Type.String(),
-    })
-  ),
-  name: Type.Optional(
-    Type.Object({
-      from: Type.String(),
-    })
-  ),
-});
-
-export const payloadSchema = Type.Object({
-  action: Type.String(),
-  issue: Type.Optional(issueSchema),
-  label: Type.Optional(labelSchema),
-  comment: Type.Optional(commentSchema),
-  sender: userSchema,
-  repository: repositorySchema,
-  organization: Type.Optional(organizationSchema),
-  installation: Type.Optional(installationSchema),
-  repositories_added: Type.Optional(Type.Array(repositorySchema)),
-  changes: Type.Optional(changesSchema),
-});
-
+export type User = RestEndpointMethodTypes["users"]["getByUsername"]["response"]["data"];
+export type Issue = RestEndpointMethodTypes["issues"]["get"]["response"]["data"];
+export type Label = RestEndpointMethodTypes["issues"]["listLabelsOnIssue"]["response"]["data"][0];
 export type HandlerReturnValuesNoVoid = null | string | LogReturn;
+
+export const issueType = {
```

Comment:
Should be an enum, and uppercase

Reply:
I've been of the belief that enums are considered bad TS practice in most cases. I've watched/listened to quite a bit of [Matt Pocock](https://www.youtube.com/@mattpocockuk) who seems to be an authority in the TS world afaik and he mentioned once (I'm paraphrasing here) that it was asked - if they could roll things back would they introduce enums and the answer was no they would have avoided it, this was someone who was involved in the development of TS I forget their name tho.

Reply:
Nothing against this syntax, but The uppercase would be needed since it's used as a type imo.

Reply:
I thought eslint would complain if we made a follow this convention my mistake

Reply:
It's a debatable topic the whole `enum` and `const enum` thing isn't it. I was surprised to see them in use across UBQ at first actually I expected everyone to be in the against enum camp

Reply:
I believe it is. I do not mind either, and I don't think the overhead is that big, but if it is mentioned in best practices for modern TS it makes sense to follow these rules. Feel free to open this topic inside of the `ts-template` repo.

Reply:
@gentlementlegen eslint does complain it just must not have updated fast enough when I checked.

> 8:14  error  Variable name `IssueType` must match one of the following formats: strictCamelCase, UPPER_CASE

This err when running `yarn format`

Reply:
> Given the context of the discussion and the specifics of TypeScript usage, here is a response and possible resolution:
> 
> 1. **Enum vs Object with `as const`:**
>    - The preference for using objects with `as const` over enums aligns with the modern TypeScript best practices. It helps keep the codebase closer to standard JavaScript and avoids the additional generated code that enums introduce. This is particularly beneficial when considering tree-shaking and overall bundle size.
>    - **Supporting Reference:** As per the TypeScript documentation, "In modern TypeScript, you may not need an enum when an object with `as const` could suffice."
> 
> 2. **Naming Convention:**
>    - The issue regarding naming conventions is valid. The ESLint rule suggests that type-like entities should follow the UPPER_CASE naming convention.
>    - To resolve this, you should update the `issueType` to `ISSUE_TYPE` to satisfy the ESLint rules without switching to an enum.
> 
> ### Updated Code
> Here's how you can update the code to adhere to the naming conventions while sticking with the object with `as const`:
> 
> ```typescript
> import { LogReturn } from "../adapters/supabase/helpers/logs";
> import { RestEndpointMethodTypes } from "@octokit/rest";
> 
> export type User = RestEndpointMethodTypes["users"]["getByUsername"]["response"]["data"];
> export type Issue = RestEndpointMethodTypes["issues"]["get"]["response"]["data"];
> export type Label = RestEndpointMethodTypes["issues"]["listLabelsOnIssue"]["response"]["data"][0];
> export type HandlerReturnValuesNoVoid = null | string | LogReturn;
> 
> export const ISSUE_TYPE = {
>   OPEN: "open",
>   CLOSED: "closed",
> } as const;
> ```
> 
> ### Summary of Points:
> - **Enum vs Object with `as const`:** Use object with `as const` to stay in line with modern TypeScript practices and keep the code closer to standard JavaScript.
> - **Naming Convention:** Change `issueType` to `ISSUE_TYPE` to comply with the ESLint naming conventions and avoid naming errors.
> 
> This approach should address both the concerns about using enums and the ESLint naming conventions.

Reply:
So the outcome then is to use `const enum` and adjust the name yeah?

Reply:
Use the code it supplied thats fine.